functions/src/export-csv.ts

```diff
@@ -167,3 +171,31 @@ function getFileName(jobName: string) {
   const fileBase = jobName.toLowerCase().replace(/[^a-z0-9]/gi, '-');
   return `${fileBase}.csv`;
 }
+
+function getAllPropertiesName(
+  lois: FirebaseFirestore.QuerySnapshot<FirebaseFirestore.DocumentData>): string[] {
+  let properties: string[] = [];
+  lois.forEach((loi) => {
+    Object.keys(loi.get('properties')).forEach((prop) => {
+      // We don't push 'id' because it's added initially in the first column of the CSV file.
+      if (!properties.includes(prop) && prop !== 'id') {
+        properties.push(prop);
+      }
+    })
+  })
+
+  return properties
+}
+
+function extractLoiProperties(
+  loi: FirebaseFirestore.QueryDocumentSnapshot<FirebaseFirestore.DocumentData>,
+  allLoiProperties: string[]): string[] {
+    let res: string[] = [];
+    if (!loi.get('properties')) {
+      throw Error(`${loi.data} does not have a 'properties' field`)
```

Comment:
Not all LOIs will have properties; we should return a null for those fields instead.

Reply:
Added empty list, does that work?